### PR TITLE
User service

### PR DESCRIPTION
**Pull request by karolinak00**
I have already implemented user service. I'm waitng for your review.

### CODE REVIEW

File: backend/src/main/java/pl/polsl/shop/user/UserService.java [hunk 1]
```diff
@@ -1,13 +1,27 @@
...
 
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
-import pl.polsl.shop.user.User;
+import pl.polsl.shop.cart.ShoppingCartService;
+import pl.polsl.shop.order.OrderService;
 
-// This service is assigned to Karolina, I only made a few necessary method signatures, so that the program would compile
 @Service("userService")
 public class UserService {
```
Comment (michloj571) [inline backend/src/main/java/pl/polsl/shop/user/UserService.java:9]: What about adding users, clearing user data and so on? Implementing theese functionalities would also require extension of the UserRepository

File: backend/src/main/java/pl/polsl/shop/user/UserException.java [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package pl.polsl.shop.user;
+
+public class UserException extends RuntimeException {
```
Comment (michloj571) [inline backend/src/main/java/pl/polsl/shop/user/UserException.java:3]: If it's the main user exception, I'd suggest to make it an abstract class

File: backend/src/main/java/pl/polsl/shop/user/Address.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
+package pl.polsl.shop.user;
+
+public class Address {
```
Comment (michloj571) [inline backend/src/main/java/pl/polsl/shop/user/Address.java:3]: I think it should be modelled to an entity and would also require its own table in the database

**Review by michloj571 — CHANGES_REQUESTED** [review]
I'd say it needs work, especially the user service lacks the necessary features

**Comment by karolinak00** [comment]
I have already introduced the changes proposed by you and completed the implementation of the user-service. I'm waiting for your review.

**Pull request merged**